### DISCUSSION TITLE
flexible job-shop taks alternative which makes other tasks obsolete

### DISCUSSION THREAD

**Discussion by alex7786** [google/or-tools#3615] [CP-SAT questions] [answered]
I'm trying to modify the [flexible job shop example](https://github.com/google/or-tools/blob/stable/examples/python/flexible_job_shop_sat.py) by adding a task alternative which, when chosen, makes the following tasks obsolete but only in that case. Extending the example jobs with a 4th alternative in task 0:
```
jobs = [  # task = (processing_time, machine_id)   
        [  # Job 0  
            [(3, 0), (1, 1), (5, 2), (8,3)],  # task 0 with 4 alternatives   
            [(2, 0), (4, 1), (6, 2)],  # task 1 with 3 alternatives
            [(2, 0), (3, 1), (1, 2)],  # task 2 with 3 alternatives
        ],
        [  # Job 1
            [(2, 0), (3, 1), (4, 2), (8,3)],
            [(1, 0), (5, 1), (4, 2)],
            [(2, 0), (1, 1), (4, 2)],
        ],
        [  # Job 2
            [(2, 0), (1, 1), (4, 2), (7,3)],
            [(2, 0), (3, 1), (4, 2)],
            [(3, 0), (1, 1), (5, 2)],
        ],
    ]
```

So as an example if machine_id 3 is chosen in task 0, all subsequent tasks would be obsolete since the job can be done entirely on machine_id 3. If another machine_id in task 0 is chosen all subsequent tasks have to be done.
Has anybody done something similar and can notch me in the right direction? My first idea was to add a boolVar which is enforced when in task 0 machine_id 3 is chosen and that forces all other presences to 0 but that did not work.

**Comment by lperron** [comment, marked as answer]
Write implications between the presence literals of each intervals.
